From c02d5cb327f34f8ea63f4ac4a327c8fe0dc3c91b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 29 Aug 2024 18:28:01 +0000
Subject: [PATCH] Fix default icon name

The icon is called 'org.gnome.Glade', not 'glade'. Also set the application
ID for prgname to ensure that Wayland compositors could match the window with
the application and show the appropriate icon for them.
---
 src/glade-window.c | 2 +-
 src/main.c         | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/glade-window.c b/src/glade-window.c
index b54beb21..0e16dac2 100644
--- a/src/glade-window.c
+++ b/src/glade-window.c
@@ -636,7 +636,7 @@ recent_add (GladeWindow *window, const gchar *path)
   recent_data->description = NULL;
   recent_data->mime_type = "application/x-glade";
   recent_data->app_name = (gchar *) g_get_application_name ();
-  recent_data->app_exec = g_strjoin (" ", g_get_prgname (), "%u", NULL);
+  recent_data->app_exec = g_strdup ("glade %u");
   recent_data->groups = NULL;
   recent_data->is_private = FALSE;
 
diff --git a/src/main.c b/src/main.c
index 019c89f5..67fe36b1 100644
--- a/src/main.c
+++ b/src/main.c
@@ -42,6 +42,7 @@
 #endif
 
 #define APPLICATION_NAME (_("Glade"))
+#define APPLICATION_ID "org.gnome.Glade"
 
 
 /* Application arguments */
@@ -92,9 +93,8 @@ startup (GApplication *application)
   glade_setup_log_handlers ();
 
   g_set_application_name (APPLICATION_NAME);
-
-  gtk_window_set_default_icon_name ("glade");
-
+  g_set_prgname (APPLICATION_ID);
+  gtk_window_set_default_icon_name (APPLICATION_ID);
 }
 
 static void
@@ -185,7 +185,7 @@ main (int argc, char *argv[])
   _init_cairo_xlib_workaround ();
 #endif
 
-  app = gtk_application_new ("org.gnome.Glade", G_APPLICATION_HANDLES_OPEN);
+  app = gtk_application_new (APPLICATION_ID, G_APPLICATION_HANDLES_OPEN);
 
   g_application_set_option_context_summary (G_APPLICATION (app),
                                             N_("Create or edit user interface designs for GTK+ or GNOME applications."));
-- 
2.49.0

